src/main/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacade.java

```diff
@@ -305,6 +307,8 @@ public final Response startQuizAttempt(@Context final Request request, @Context
             return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR, message).toResponse();
         } catch (AttemptCompletedException e) {
             return new SegueErrorResponse(Status.FORBIDDEN, "You have already completed your attempt at this quiz.").toResponse();
+        } catch (AssignmentCancelledException e) {
+            return new SegueErrorResponse(Status.GONE, "This quiz assignment has been cancelled.").toResponse();
```

Comment:
See my other comment, but we've never needed a 410 Gone before because we've done the filtering of deleted things in the database so they have no side effects in the code; it means we can't tell the difference between deleted and non-existent but I think that's fine.

Reply:
So, I went for the `GONE` because we send out links as soon as teacher sets a task, so I expected to use the status in the front-end to give a nicer message when that occurs instead of a generic site error or SegueDatabaseException message.